### ISSUE TITLE
Avatar performance improvements

### ISSUE THREAD

**Issue by rullzer** [owncloud/core#22414]
Curerntly we just store the avatars in `/<user>/avatar.jpg` (or `<user>/avatar.png`). While this is simple it also has certain drawbacks. Like the bug we hit with https://github.com/owncloud/core/issues/22119

Further more it would be great to also get avatars of trusted ownClouds locally since we have the vcards anyway.
1. We should rethink where and how we store avatars (we should still cache generated avatars).
2. We should actively start caching avatars client side
   - Allow caching of avatars... at least checking validating by etag
   - Your own avatar should be very limited cached since you can update that and want that to be reflected fast
   - Other users avatars can be cached much longer
3. Our avatar javascript code has become somewhat big and unefficient.
   - We should start using promised (so each avatar is really only checked once per page load).

So basically we should rethink our avatar handling in general.

CC: @PVince81 @DeepDiver1975

**Comment by butonic** [comment]
Isn't the avatar something like a thumbnail?

Or should we store it in a dedicated location?

Maybe related https://github.com/owncloud/core/issues/25213

**Comment by PVince81** [comment]
I'd suggest storing it outside of "/files" so we don't need a full FS setup just to access it.
There are other related issues I met due to syncjob + ldap + avatar + email: https://github.com/owncloud/core/issues/24423#issuecomment-226525633

Maybe a proper location would be "data/$user/avatar", then if there are multiple sizes they can just be stored there.

**Comment by PVince81** [comment]
I'm setting this to 9.2 because this have more consequences on performance and other nasty side effects.

**Comment by DeepDiver1975** [comment]
> I'm setting this to 9.2 because this have more consequences on performance and other nasty side effects.

agreed

**Comment by PVince81** [comment]
Hmmm, just now I thought about modifying `initMountsPoints` to receive a second argument `$homeOnly` but realized it won't cover the cases where a cron job runs and wouldn't tear down all the FS. So the best solution is rather to invest time into removing the avatars from the filesystem layer and finding a way to access the avatar list that doesn't require a FS setup.

**Comment by PVince81** [comment]
PR to move avatars: https://github.com/owncloud/core/pull/25790

**Comment by PVince81** [comment]
- [ ] repair steps
- [ ] unit tests

**Comment by PVince81** [comment]
Is the solution from 9.2 that moves the avatars to separate folders already enough to move this ticket to backlog and decrease severity @DeepDiver1975 @butonic ?

**Comment by PVince81** [comment]
Decreasing severity to high

**Comment by PVince81** [comment]
moving to backlog for now.

Time will tell whether the 10.0 solution of having a separate avatar folder is enough to reduce the severity of this issue. Additional tasks are still valid to further improve avatar handling.

**Comment by SamuAlfageme** [comment]
Also linking https://github.com/owncloud/core/pull/26872 here in case it contributes to decreasing severity and/or close.
